localization/ndt_scan_matcher/include/ndt_scan_matcher/map_update_module.hpp

```diff
@@ -88,6 +91,9 @@ class MapUpdateModule
   // Indicate if there is a prefetch thread waiting for being collected
   NdtPtrType secondary_ndt_ptr_;
   bool need_rebuild_;
+  std::mutex last_update_pos_mtx_;
+  // A copy of last update position, used for checking map range
+  std::optional<geometry_msgs::msg::Point> last_update_pos_;
```

Comment:
I'm confused about the need to add this variable. Why is not it enough to just use `last_update_position_`?

Reply:
@SakodaShintaro I removed the second last_update_pos_. The mutex is now used for last_update_position_.